plugins/BEdita/Core/config/Migrations/20171109094654_AddDefaultPropertyTypes.php

```diff
@@ -0,0 +1,106 @@
+<?php
+use Migrations\AbstractMigration;
+
+/**
+ * Add default property types.
+ */
+class AddDefaultPropertyTypes extends AbstractMigration
+{
+
+    /**
+     * {@inheritDoc}
+     */
+    public function up()
+    {
+        $this->table('properties')
+            ->removeColumn('multiple')
+            ->removeColumn('options_list')
+            ->removeColumn('list_view')
+            ->update();
+
+        $this->table('properties')
+            ->addColumn('is_nullable', 'boolean', [
+                'after' => 'label',
+                'comment' => 'is property nullable?',
+                'default' => '1',
+                'length' => null,
+                'null' => false,
+            ])
+            ->update();
+
+        $this->table('property_types')
+            ->insert([
+                [
+                    'name' => 'string',
+                    'params' => '{"type":"string"}',
+                ],
+                [
+                    'name' => 'text',
+                    'params' => '{"type":"string","contentMediaType":"text/html"}',
+                ],
+                [
+                    'name' => 'status',
+                    'params' => '{"type":"string","enum":["on","off","draft"]}',
+                ],
+                [
+                    'name' => 'email',
+                    'params' => '{"type":"string","format":"email"}',
+                ],
+                [
+                    'name' => 'url',
+                    'params' => '{"type":"string","format":"uri"}',
+                ],
+                [
+                    'name' => 'date',
+                    'params' => '{"type":"string","format":"date-time"}',
+                ],
+                [
+                    'name' => 'number',
```

Comment:
what about `integer` as core type `{"type":"number", "format":"integer"}` ?

Reply:
It seems a good idea. What about specs? Anyway I will eventually open another issue for that-

Reply:
in draft6 and 7 there's no `integer` primitive type but only `number`
http://json-schema.org/latest/json-schema-core.html#rfc.section.4.2.1
but `integer` is one of the most obvious custom type

Reply:
`integer` is an allowed `type`. However, it might be difficult to validate since some programming languages just don't distinguish between numbers with and without a fractional part.

We might add `{"type": "integer", "multipleOf": 1}` to try to achieve the maximum portability… 🤔

Reply:
@stefanorosanelli According to [Draft-7 JSON Schema Validation](http://json-schema.org/latest/json-schema-validation.html#rfc.section.6.1.1) "integer" is an allowed keyword for `type`…

Reply:
let's use simply `{"type": "integer"}` for now then